### PR TITLE
test(module: drawer): Test, refactor and document Drawer

### PR DESCRIPTION
**Pull request by kooliokey**
Bug fix:
- Content string would never render (wrong condition checked)

Refactors:
- Remove `this.`
- Remove code that didn't do anything helpful from my testing - will comment on inline
- Moved code to make it more obvious only one condition can exist at a time - will comment on inline

Documentation site update to match available parameters

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [X] Refactoring
- [ ] Code style optimization
- [X] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#2644 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix bug where Content wouldn't render in Drawer if it was a string and not RenderFragment       |
| 🇨🇳 Chinese |  修复如果内容是字符串而不是 RenderFragment，则内容不会在 Drawer 中呈现的错误         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/drawer/Drawer.razor.cs [hunk 1]
```diff
@@ -356,41 +360,32 @@ private string Transform
...
-            this.SetClass();
+            SetClass();
 
             base.OnInitialized();
         }
 
         protected override void OnParametersSet()
         {
-            this.SetClass();
-            if (string.IsNullOrEmpty(Placement) && Placement != _originalPlacement)
```
Comment (kooliokey) [inline components/drawer/Drawer.razor.cs:385]: This block and line 448 block that was removed didn't do anything when Placement was null or empty string. I tested with a without it and it behaved the same both ways. 

I checked when it was added to the repo and it was from the start and also moved in a refactor to fix a bug. I double checked that bug and it is still fixed with this removed. It was related to multiple drawers open.

File: components/drawer/Drawer.razor.cs [hunk 2]
```diff
@@ -502,37 +482,16 @@ private async Task HandleClose(bool isChangeByParamater = false)
             }
         }
 
-        private void CalcAnimation()
```
Comment (kooliokey) [inline components/drawer/Drawer.razor.cs:505]: I moved this to be inline with where it was called to make it obvious only the width or height transition could exist at once due to it being dependent on Placement's value. It is moved to line 490 below

File: tests/AntDesign.TestKit/AntDesignTestBase.cs [hunk 3]
```diff
@@ -22,6 +22,7 @@ public AntDesignTestBase(bool useMoq = false)
             //Needed for Tests using Overlay
             Services.AddScoped<AntDesign.JsInterop.DomEventService>(sp => new TestDomEventService(Context.JSInterop.JSRuntime, MockedDomEventListener));
             JSInterop.SetupVoid(JSInteropConstants.OverlayComponentHelper.DeleteOverlayFromContainer, _ => true);
+            JSInterop.Mode = JSRuntimeMode.Strict;
```
Comment (kooliokey) [inline tests/AntDesign.TestKit/AntDesignTestBase.cs:25]: Added this so even if a test changes it, all other tests still have it set to the default Strict. Not 100% sure the lifetime of this service but figured it was a good precaution.

**Comment by kooliokey** [comment]
> # [Codecov](https://codecov.io/gh/ant-design-blazor/ant-design-blazor/pull/2833?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ant-design-blazor) Report
> Base: **43.08**% // Head: **42.38**% // Decreases project coverage by **`-0.70%`** ⚠️
> 
> > Coverage data is based on head [(`7d4009f`)](https://codecov.io/gh/ant-design-blazor/ant-design-blazor/pull/2833?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ant-design-blazor) compared to base [(`f9c5f5c`)](https://codecov.io/gh/ant-design-blazor/ant-design-blazor/commit/f9c5f5cdf3d80da98ebb56564324c504363f5228?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ant-design-blazor).
> > Patch coverage: 86.20% of modified lines in pull request are covered.
> 
> Additional details and impacted files
> [☔ View full report at Codecov](https://codecov.io/gh/ant-design-blazor/ant-design-blazor/pull/2833?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ant-design-blazor). 📢 Do you have feedback about the report comment? [Let us know in this issue](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ant-design-blazor).

How did coverage go _down_? 😢

**Comment by ElderJames** [comment]
Yes, that's a weird statistic. It feels different than the statistics of the master branch

File: components/drawer/Drawer.razor.cs [hunk 4]
```diff
@@ -442,33 +439,16 @@ protected override async Task OnAfterRenderAsync(bool isFirst)
         }
 
         private Timer _timer;
-        private int _zIndex = 1000;
+        private int _zIndex = DefaultZIndez;
         private string _zIndexStyle = "";
 
-        private void TriggerPlacementChangeCycleOnce()
```
Comment (ElderJames) [inline components/drawer/Drawer.razor.cs:448]: @zxyao145 Please check if this method is useless.

Comment (zxyao145) [reply components/drawer/Drawer.razor.cs:448]: Yes, this method will only be triggered when the Placement changes (line 394).

Comment (ElderJames) [reply components/drawer/Drawer.razor.cs:448]: Thanks @zxyao145 , please help to confirm whether there is any problem after deletion.

**Review by zxyao145 — APPROVED** [review]

**Review by ElderJames — APPROVED** [review]

**Pull request merged**